JustSaying/Messaging/PublishMetadata.cs

```diff
@@ -0,0 +1,31 @@
+using System;
+using System.Collections.Generic;
+
+namespace JustSaying.Messaging
+{
+    public class PublishMetadata
+    {
+        public TimeSpan? Delay { get; set; }
+
+        public IDictionary<string, MessageAttributeValue> MessageAttributes { get; set; }
+
+        public void AddMessageAttribute(string key, MessageAttributeValue value)
```

Comment:
Would be nice if this (and the other) method returned `PublishMetadata` so you can chain fluently in your setup.

```csharp
var metadata = new AddMessageAttribute()
    .AddMessageAttribute("foo", "bar")
    .AddMessageAttribute("qux", "zab");
```